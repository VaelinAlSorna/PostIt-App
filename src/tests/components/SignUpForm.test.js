import React from 'react';
import { shallow } from 'enzyme';
import SignUpForm from '../../components/SingUpForm';

let changeForm, onSubmit, wrapper;

beforeEach(() => {
    changeForm = jest.fn();
    onSubmit = jest.fn();
    wrapper = shallow(<SignUpForm 
        authFailed={null} 
        changeForm={changeForm} 
        onSubmit={onSubmit}
    />)
})

test('should render SignUpForm correctly', () => {
    expect(wrapper).toMatchSnapshot();
});


test('should call this.props.onSubmit when form is submitted', () => {
    const name = 'Mark';
    const password = 'heslo1';
    const passwordAgain = 'heslo1';
    wrapper.setState({ name, password, passwordAgain  });
    wrapper.find('form').simulate('submit' ,{
        preventDefault: () => {}
    });
    expect(onSubmit).toHaveBeenLastCalledWith({name, password});
});

test('should not call this.props.onSubmit if name or passwords do not have length greater than 3', () => {
    const name = 'Mark';
    const password = 'heslo';
    const passwordAgain = 'heslo';
    wrapper.setState({ name, password, passwordAgain  });
    wrapper.find('form').simulate('submit' ,{
        preventDefault: () => {}
    });
    expect(wrapper.state('error')).toBe('Your name must contain at least 3 characters and password must contain at least 6');
});

test('should not call this.props.onSubmit if passwords do not mathc', () => {
    const name = 'Mark';
    const password = 'heslo1';
    const passwordAgain = 'heslo2';
    wrapper.setState({ name, password, passwordAgain  });
    wrapper.find('form').simulate('submit' ,{
        preventDefault: () => {}
    });
    expect(wrapper.state('error')).toBe('Passwords do not match!');
});

test('should change name on input change', () => {
    const value = 'Mark';
    wrapper.find('input').at(0).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('name')).toBe(value);
});

test('should change password on input change', () => {
    const value = 'heslo1';
    wrapper.find('input').at(1).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('password')).toBe(value);
});

test('should call changeForm when clicked on OR SignUp button', () => {
    wrapper.find('button').simulate('click');
    expect(changeForm).toHaveBeenCalled();
});

test('should render error when authFailed is passed down with value', () => {
    wrapper = shallow(<SignUpForm 
        authFailed={{ error: 'This name already exists', login: false}} 
        changeForm={changeForm} 
        onSubmit={onSubmit}
    />)
    expect(wrapper).toMatchSnapshot();
});
