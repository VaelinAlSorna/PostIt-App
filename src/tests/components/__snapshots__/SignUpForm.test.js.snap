// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SignUpForm correctly 1`] = `
<div
  className="not-logged-in"
>
  <form
    className="not-logged-in__group form"
    onSubmit={[Function]}
  >
    <div
      className="form__group"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="form__input"
        id="name"
        onChange={[Function]}
        placeholder="Your name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form__group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form__input"
        id="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form__group u-margin-bottom-large"
    >
      <label
        htmlFor="password"
      >
        Repeat Password
      </label>
      <input
        className="form__input"
        onChange={[Function]}
        placeholder="Repeat Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <input
      className="btn btn--green"
      type="submit"
      value="Sign Up"
    />
  </form>
  <p
    className="not-logged-in__group"
  >
    OR
  </p>
  <button
    className="not-logged-in__group btn btn--white"
    onClick={[Function]}
  >
    Log In
  </button>
  <p />
</div>
`;

exports[`should render error when authFailed is passed down with value 1`] = `
<div
  className="not-logged-in"
>
  <form
    className="not-logged-in__group form"
    onSubmit={[Function]}
  >
    <div
      className="form__group"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="form__input"
        id="name"
        onChange={[Function]}
        placeholder="Your name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form__group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form__input"
        id="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form__group u-margin-bottom-large"
    >
      <label
        htmlFor="password"
      >
        Repeat Password
      </label>
      <input
        className="form__input"
        onChange={[Function]}
        placeholder="Repeat Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <input
      className="btn btn--green"
      type="submit"
      value="Sign Up"
    />
  </form>
  <p
    className="not-logged-in__group"
  >
    OR
  </p>
  <button
    className="not-logged-in__group btn btn--white"
    onClick={[Function]}
  >
    Log In
  </button>
  <p
    className="auth-error"
  >
    This name already exists
  </p>
</div>
`;
